src/test/java/org/fogbowcloud/manager/core/OrderControllerTest.java

```diff
@@ -222,91 +241,136 @@ public void testGetUserAllocationInvalidInstanceType() throws UnexpectedExceptio
-        Assert.assertEquals(OrderState.CLOSED, test.getOrderState());
+        Order order = this.closedOrdersList.getNext();
+        this.failedOrdersList.resetPointer();
+
+        // verify
+        Assert.assertNull(this.failedOrdersList.getNext());
+        Assert.assertNotNull(order);
+        Assert.assertEquals(computeOrder, order);
+        Assert.assertEquals(OrderState.CLOSED, order.getOrderState());
     }
 
+    // test case: Checks if deleting a fulfiled order, this one will be moved to the closed orders list.
     @Test
     public void testDeleteOrderStateFulfilled() throws UnexpectedException, OrderNotFoundException {
+        // set up
         String orderId = getComputeOrderCreationId(OrderState.FULFILLED);
         ComputeOrder computeOrder = (ComputeOrder) this.activeOrdersMap.get(orderId);
 
+        // verify
+        Assert.assertNotNull(this.fulfilledOrdersList.getNext());
         Assert.assertNull(this.closedOrdersList.getNext());
 
+        // exercise
         this.ordersController.deleteOrder(computeOrder);
 
         Order test = this.closedOrdersList.getNext();
+
+        // verify
+        Assert.assertNull(this.fulfilledOrdersList.getNext());
         Assert.assertNotNull(test);
         Assert.assertEquals(computeOrder, test);
         Assert.assertEquals(OrderState.CLOSED, test.getOrderState());
     }
 
+    // test case: Checks if deleting a spawning order, this one will be moved to the closed orders list.
     @Test
     public void testDeleteOrderStateSpawning() throws UnexpectedException, OrderNotFoundException {
+        // set up
         String orderId = getComputeOrderCreationId(OrderState.SPAWNING);
         ComputeOrder computeOrder = (ComputeOrder) this.activeOrdersMap.get(orderId);
 
+        // verify
+        Assert.assertNotNull(this.spawningOrdersList.getNext());
         Assert.assertNull(this.closedOrdersList.getNext());
 
+        // exercise
         this.ordersController.deleteOrder(computeOrder);
 
         Order test = this.closedOrdersList.getNext();
+
+        // verify
+        Assert.assertNull(this.spawningOrdersList.getNext());
         Assert.assertNotNull(test);
         Assert.assertEquals(computeOrder, test);
         Assert.assertEquals(OrderState.CLOSED, test.getOrderState());
     }
 
+    // test case: Checks if deleting a pending order, this one will be moved to the closed orders list.
     @Test
     public void testDeleteOrderStatePending() throws UnexpectedException, OrderNotFoundException {
+        // set up
         String orderId = getComputeOrderCreationId(OrderState.PENDING);
         ComputeOrder computeOrder = (ComputeOrder) this.activeOrdersMap.get(orderId);
 
+        // verify
+        Assert.assertNotNull(this.pendingOrdersList.getNext());
         Assert.assertNull(this.closedOrdersList.getNext());
 
+        // exercise
         this.ordersController.deleteOrder(computeOrder);
 
         Order test = this.closedOrdersList.getNext();
+
```

Comment:
lines 332 and 334 do not belong to the exercise block. Please, check the other test cases